src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -9,342 +11,367 @@
 						{
-							var dir = Path.GetDirectoryName(_resolveAssemblyLocation);
-							if (!string.IsNullOrEmpty(dir))
+							var relPath = Path.Combine(dir, assemblyName.Name + ".dll");
+							if (File.Exists(relPath))
 							{
-								var relPath = Path.Combine(dir, assemblyName.Name + ".dll");
-								if (File.Exists(relPath))
+								if (this.Log().IsEnabled(LogLevel.Trace))
 								{
-									if (this.Log().IsEnabled(LogLevel.Trace))
-									{
-										this.Log().LogTrace("Loading assembly from resolved path: {relPath}", relPath);
-									}
-
-									using var fs = File.Open(relPath, FileMode.Open, FileAccess.Read, FileShare.Read);
-									return context.LoadFromStream(fs);
+									this.Log().LogTrace("Loading assembly from resolved path: {relPath}", relPath);
 								}
-							}
-						}
-						catch (Exception exc)
-						{
-							if (this.Log().IsEnabled(LogLevel.Error))
-							{
-								this.Log().LogError(exc, "Failed for load dependency: {assemblyName}", assemblyName);
+
+								using var fs = File.Open(relPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+								return context.LoadFromStream(fs);
 							}
 						}
 					}
-					else
+					catch (Exception exc)
 					{
-						if (this.Log().IsEnabled(LogLevel.Debug))
+						if (this.Log().IsEnabled(LogLevel.Error))
 						{
-							this.Log().LogDebug("Failed for identify location of dependency: {assemblyName}", assemblyName);
+							this.Log().LogError(exc, "Failed for load dependency: {assemblyName}", assemblyName);
 						}
 					}
-
-					// We haven't found the assembly in our context, let the runtime
-					// find it using standard resolution mechanisms.
-					return null;
-				};
-
-				if (!_loadContexts.TryAdd(applicationId, (loadContext, 1)))
+				}
+				else
 				{
-					if (this.Log().IsEnabled(LogLevel.Trace))
+					if (this.Log().IsEnabled(LogLevel.Debug))
 					{
-						this.Log().LogTrace("Failed to add a LoadContext for : {appId}", applicationId);
+						this.Log().LogDebug("Failed for identify location of dependency: {assemblyName}", assemblyName);
 					}
 				}
 
-				return loadContext;
+				// We haven't found the assembly in our context, let the runtime
+				// find it using standard resolution mechanisms.
+				return null;
+			};
+
+			if (!_loadContexts.TryAdd(applicationId, (loadContext, 1)))
+			{
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().LogTrace("Failed to add a LoadContext for : {appId}", applicationId);
+				}
 			}
+
+			return loadContext;
 		}
+	}
 
-		private void RegisterProcessor(IServerProcessor hotReloadProcessor)
-			=> _processors[hotReloadProcessor.Scope] = hotReloadProcessor;
+	private void RegisterProcessor(IServerProcessor hotReloadProcessor)
+		=> _processors[hotReloadProcessor.Scope] = hotReloadProcessor;
 
-		public async Task Run(WebSocket socket, CancellationToken ct)
-		{
-			_socket = socket;
+	public IdeChannelServer? IDEChannelServer => _ideChannelServer;
```

Comment:
Should we make this async?

Reply:
For now, no, since it may not be available if the IDE is not present (vs code for instance).